client/src/featureform/client.py

```diff
@@ -62,6 +63,7 @@ def dataframe(
             source (Union[SourceRegistrar, LocalSource, SubscriptableTransformation, str]): The source or transformation to compute the dataframe from
             variant (str): The source variant; defaults to a Docker-style random name and is ignored if source argument is not a string
             limit (int): The maximum number of records to return; defaults to NO_RECORD_LIMIT
+            asynchronous (bool): Whether to wait for the apply to finish; defaults to False to ensure sources are ready
```

Comment:
this message can be confusing. I would assume if I provide it True then it would wait.

Reply:
I agree. I was mirroring the naming and messaging in `apply`, but I can improve both to make things clearer.